src/tool/cpp/cppwinrt/strings/base_std_async_operation_with_progress.h

```diff
@@ -4,65 +4,32 @@ WINRT_EXPORT namespace std::experimental
     template <typename TResult, typename TProgress, typename... Args>
     struct coroutine_traits<winrt::Windows::Foundation::IAsyncOperationWithProgress<TResult, TProgress>, Args...>
     {
-        struct promise_type final : winrt::impl::promise_base<promise_type, winrt::Windows::Foundation::IAsyncOperationWithProgress<TResult, TProgress>,
+        struct promise_type final : winrt::impl::promise_base<promise_type,
+            winrt::Windows::Foundation::IAsyncOperationWithProgress<TResult, TProgress>,
             winrt::Windows::Foundation::AsyncOperationWithProgressCompletedHandler<TResult, TProgress>, TProgress>
         {
-            using AsyncStatus = winrt::Windows::Foundation::AsyncStatus;
             using ProgressHandler = winrt::Windows::Foundation::AsyncOperationProgressHandler<TResult, TProgress>;
 
-            void Progress(ProgressHandler const& handler)
+            void Progress(ProgressHandler const& handler) noexcept
             {
                 winrt::slim_lock_guard const guard(this->m_lock);
                 m_progress = handler;
             }
 
-            ProgressHandler Progress()
+            ProgressHandler Progress() noexcept
             {
                 winrt::slim_lock_guard const guard(this->m_lock);
                 return m_progress;
             }
 
-            TResult GetResults()
+            TResult get_return_value() const noexcept
             {
-                winrt::slim_lock_guard const guard(this->m_lock);
-
-                if (this->m_status == AsyncStatus::Completed)
-                {
-                    return m_result;
-                }
-
-                this->rethrow_if_failed();
-                WINRT_ASSERT(this->m_status == AsyncStatus::Started);
-                throw winrt::hresult_illegal_method_call();
+                return m_result;
             }
 
-            void return_value(TResult const& result)
+            void return_value(TResult const& result) noexcept
```

Comment:
`noexcept` assumes that `m_result = result` never throws. Is this valid assumption?

Reply:
Same as above. Since result is guaranteed to be a WinRT type and WinRT types in C++/WinRT all provide noexcept copy semantics.